server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1037,7 +1054,7 @@ public void apply(T value, Settings current, Settings previous) {
         return new Setting<>(key, fallback, parser, properties);
     }
 
-    public static Setting<String> simpleString(String key, Validator<String> validator, Property... properties) {
+    public static Setting<String> simpleStringWithValidator(String key, Validator<String> validator, Property... properties) {
```

Comment:
I think we could leave this called `simpleString`. The convention seems to be not to call out the existence of the Validator explicitly in other calls. I see that this might cause some issues with overloading but I think they're surmountable.

Reply:
Exactly, I'll try to work around overloading issues.

Reply:
The build is setup to disallow any compiler warning.

I've tried to work around this issue by having `simpleString` API with different number of functional interface parameters, but the warning is still there and I don't understand why. I've posted a [question on SO](https://stackoverflow.com/questions/53080812/java-functional-interfaces-weird-potentially-ambiguous-warning) to get some feedback on this issue, I'll let you know.